crates/ruma-client-api/src/r0/account/register.rs

```diff
@@ -72,7 +72,8 @@ ruma_api! {
         ///
         /// [admin]: https://matrix.org/docs/spec/application_service/r0.1.2#server-admin-style-permissions
         #[serde(rename = "type", skip_serializing_if = "Option::is_none")]
-        pub login_type: Option<&'a LoginType>,
+        pub login_type: Option<&'a LoginType>, // TODO: should this field be gated
+        // by the app-service feature?
```

Comment:
The docs/spec seem to say that this is only used in the appservice-api should this be gated? The same would go for the `LoginType` below.